### ISSUE TITLE
Add option for "no-shadowed-variable" to ignore parameter decorators

### ISSUE THREAD

**Issue by lonix1** [palantir/tslint#4582]
Problem code:

````typescript
public foo(
  @request request: express.Request,    // Shadowed name: 'request' (no-shadowed-variable)
  @response response: express.Response, // Shadowed name: 'response' (no-shadowed-variable)
  @next next: express.NextFunction      // Shadowed name: 'next' (no-shadowed-variable)
) {
  // .......
}
````
The decorators and the argument names are the same, so the rule is triggered.

Options:
:-1: ignore the problem
:-1: disable the rule 
:-1: rename the argument (those names are canonical so this is a bad idea) 
:+1: add an option to the rule to handle this case

**Comment by adidahiya** [comment]
TIL about parameter decorators :)

**Comment by lonix1** [comment]
It's from the [inversify-express-utils](https://github.com/inversify/inversify-express-utils#step-1-decorate-your-controllers) package.

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
